plugins/Files/js/sagas/helpers.js

```diff
@@ -105,7 +105,19 @@ export const ls = (files, path) => {
 			type,
 		})
 	})
-	return parsedFiles.toList().sortBy((file) => file.name)
+	return parsedFiles.toList().sortBy((file) => file.name).sort(directoriesFirst)
```

Comment:
note that because two file types, `directory` and `file` happen to be lexicographically in the correct order for our desired behavior, we could do a `.sortBy((file) => file.type)` here, but I decided that this was to clever.